packages/components/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useCallback } from "react";
+import { Portal } from "react-portal";
+import Button from "../Button";
+import { Cross1Icon } from "@radix-ui/react-icons";
+import { styled } from "../../stitches.config";
+
+const CloseIcon = styled(Cross1Icon, {
+  position: "absolute",
+});
+const ModalBackdrop = styled("div", {
+  position: "fixed",
+  backgroundColor: "#32484180",
+  zIndex: -1,
+  inset: 0,
+});
+const ModalRoot = styled("div", {
+  position: "fixed",
+  zIndex: 1300,
+  inset: 0,
+});
+const ModalContainer = styled("div", {
+  height: "100%",
+  display: "flex",
+  justifyContent: "center",
+  alignItems: "center",
+});
+const ModalBody = styled("div", {
+  maxWidth: 600,
+  minWidth: 400,
+  margin: 32,
+  boxShadow: "0px 12px 50px -24px rgba(0, 0, 0, 0.24);",
+  background: "$sand",
+});
+const ModalHeader = styled("div", {
+  height: 32,
+  backgroundColor: "$forest",
+  paddingTop: 12,
+  paddingLeft: 40,
+  paddingRight: 16,
+  paddingBottom: 12,
+  display: "flex",
+  justifyContent: "space-between",
+  alignItems: "center",
+});
+const ModalTitle = styled("span", {
+  fontSize: "$lg",
+  fontWeight: 600,
+  fontFamily: "$mono",
+  color: "$sand",
+});
+const CloseButton = styled(Button, {
+  height: "32px !important",
```

Comment:
Seems like the Button's `type` styles overrides any input styles. Is there a way to change that?

Reply:
It may be worth actually building out an `icon` type/variant to the button component. I would assume we'll need lots of icon-only buttons over time.

Reply:
Oh that's a good idea, I'll add it as part of this PR